### PR TITLE
fixed hdf5 observables in all-particle driver

### PR DESCRIPTION
**Pull request by Paul-St-Young**
also enabled substep in all-particle VMC

### CODE REVIEW

**Comment by Paul-St-Young** [comment]
fixed bug from #419

**Comment by Paul-St-Young** [comment]
I cleaned the VMCUpdateAll.cpp file. @jngkim and I revamped the `advanceWalker` functions to allow substep. I was a bit heavy-handed with the comments, please let me know if some of it needs to be trimmed.

I did not test `advanceWalkerForEE` because I do not have an example for it. I hope the CI has tests for this method.

**Comment by markdewing** [comment]
okay to test

File: src/QMCDrivers/VMC/VMCUpdateAll.cpp [hunk 1]
```diff
@@ -35,139 +35,55 @@ VMCUpdateAll::~VMCUpdateAll()
...
-    if (RandomGen() > g)
-    {
-      thisWalker.Age++;
-      ++nReject;
-      H.rejectedMove(W,thisWalker);
+  /* thisWalker.R will track the last accepted configuration
+   * W.R will track the proposed configuration 
+   *
+   * upon Call:
+   *   thisWalker.R,G,L,DistTables,SK,Properties must be consistent
```
Comment (ye-luo) [inline src/QMCDrivers/VMC/VMCUpdateAll.cpp]: I think thisWalker doesn't carry DistanceTables and SK.

Comment (Paul-St-Young) [reply src/QMCDrivers/VMC/VMCUpdateAll.cpp]: @ye-luo that is a good point, I was confused. 

It seems that ParticleSet::loadWalker transfers R,G,L from thisWalker to W, but only update W.DistTables or SK for pbyp driver. I think this means the internal states of W become inconsistent after calling saveWalker in the wbyw driver. 

If there a reason why the `if (pbyp)` has to be in `ParticleSet::loadWalker`?

Comment (ye-luo) [reply src/QMCDrivers/VMC/VMCUpdateAll.cpp]: This is necessary.
In pbyp moves, e-I distance table need to be up-to-date only when JeeI is used. SK need to be up-to-date if SK is needed at every pbyp move.
In allp moves, nothing is needed. Only update distance table and SK for proposed moves, if the last move is rejected, update everything with the old configuration before Hamiltonian and estimators.

At this line, please say `thisWalker.R,G,Properties must be consistent`

File: src/QMCDrivers/VMC/VMCUpdateAll.cpp [hunk 2]
```diff
@@ -35,139 +35,55 @@ VMCUpdateAll::~VMCUpdateAll()
...
+   *   recompute flag is for pbyp driver and is not used here
+   *
+   * upon Return:
+   *   thisWalker.R,G,L,DistTables,SK,Properties must be kept consistent
```
Comment (ye-luo) [inline src/QMCDrivers/VMC/VMCUpdateAll.cpp]: At this line, please say `thisWalker.R,G,Properties must be kept consistent`

**Comment by ye-luo** [comment]
@prckent if no more changes are needed, this is good to merge.

**Comment by prckent** [comment]
Thanks all.

**Pull request merged**